### ISSUE TITLE
bitbucket_pipeline_schedule resource throwing 404

### ISSUE THREAD

**Issue by alaminopu** [DrFaust92/terraform-provider-bitbucket#120]
Creating a pipeline schedule using `bitbucket_pipeline_schedule` is failing. 

```bash
resource "bitbucket_pipeline_schedule" "scan_ecr_images" {
  workspace    = "<workspace>"
  repository   = "<repo>"
  cron_pattern = "0 0 * * *"
  enabled      = true

  target {
    ref_name = "master"
    ref_type = "branch"

    selector {
      type    = "custom"
      pattern = "external-amazon-images"
    }
  }
}
```

### Terraform Version
1.3.6

### Affected Resource(s)
bitbucket_pipeline_schedule

### Debug Output
https://gist.github.com/alaminopu/f06c1c8638d84a14d9c67dd5f8582116

**Comment by alaminopu** [comment]
@DrFaust92 Could you please take a look at this issue?

**Comment by mauriceackel** [comment]
We are facing a similar issue!

@alaminopu, have you been able to resolve this? I noticed that you are using a 5-segment cron-string while bitbucket expects 7 segments (i.e. seconds and years). However, even with a 7-segment cron-string, we get a 404 although the API call works with CURL

**Comment by alaminopu** [comment]
@mauriceackel no. Wasn't able to resolve it. We are currently using manual schedule setup until this is fixed.

**Comment by DrFaust92** [comment]
Hi all,

I’ll try to take a look later this week

Thanks,

**Comment by mauriceackel** [comment]
I did some more digging and found out, that the bitbucket go client, that is used by this provider, seems to be working fine.

I tested the main branch of the client library with this setup:
```
        selector := &bitbucket.PipelineSelector{
		Type_:   "custom",
		Pattern: "mypipeline",
	}
	target := &bitbucket.PipelineSchedulePostRequestBodyTarget{
		Type_:    "pipeline_ref_target",
		RefName:  "main",
		RefType:  "branch",
		Selector: selector,
	}
	schedule := &bitbucket.PipelineSchedulePostRequestBody{
		Enabled:     true,
		CronPattern: "0 0 6-18 ? * 1-5 *",
		Target:      target,
	}
```

As this was working, I suspect the issue is somewhere in here: https://github.com/DrFaust92/terraform-provider-bitbucket/blob/master/bitbucket/resource_pipeline_schedule.go#L97-L104

**Comment by mauriceackel** [comment]
I don't have experience with Go, but I just stumbled upon this line [here](https://github.com/DrFaust92/terraform-provider-bitbucket/blob/b9109557fd9147c2850b3e92f2e989947913bdfa/bitbucket/resource_pipeline_schedule.go#L200).

This casts (at least from my understanding), the whole "target" object to a map of strings. However, it does not only consist of strings, as "selector" is again a list. So maybe this is causing the issue, but I am really unsure.
